src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // делим дату на части
+  const delimiter = date.split(fromFormat[3]);
+  const newDateObject = {};
+
+  for (let i = 0; i < 3; i++) {
+    newDateObject[fromFormat[i]] = delimiter[i];
+  }
+  // Новые переменные для преобразования годов
+
+  const CURRENT_CENTURY_THRESHOLD = 30;
+  const CENTURY_1900 = '19';
+  const CENTURY_2000 = '20';
+  // проверка если в fromFormat есть YYYY, а в toFormat — YY,
+  // то newDateObject['YY'] будет присвоено последние две цифры
```

Comment:
Remove comment. Try to write obvious code that explains all we need without comments)